api/services/namespace.go

```diff
@@ -77,9 +77,11 @@ func (s *service) CreateNamespace(ctx context.Context, req *requests.NamespaceCr
 		return nil, NewErrUserNotFound(req.UserID, err)
 	}
 
-	// When MaxNamespaces is less than or equal to zero, it means that the user has no limit
-	// of namespaces.
-	if user.MaxNamespaces > 0 {
+	// When MaxNamespaces is less than zero, it means that the user has no limit
+	// of namespaces. If the value is zero, it means he has no right to create a new namespace
+	if user.MaxNamespaces == 0 {
+		return nil, NewErrNamespaceZeroReached(user.MaxNamespaces, nil)
```

Comment:
Why a new error and not the `ErrNamespaceLimitReached`?

Reply:
Well, since zero value, is a new type of limit, is not a limit, but a condition, if is zero, is now not able to create (on purpose, not by accident) a new namespace, but with values with has a limit, for instance '3', so this implies that the plan needs to be upgraded and is not, a premeditated intention, with that in mind, I believed that adding a new error message, could resolve significant less trouble shooting with this new logic on the already integer of `MaxNamespaces`, as well as clarify that the reason for any zero value error, is doe to zero value, and not a limit, as is not actually the case

Reply:
so this clarifies well the idea that zero value is on propose and not by accident

Reply:
I would even use a more specific error mentioning that the user is not allowed to create namespaces. Being zero is just an implementation detail, but the error should be meaningful about the purpose of it.